src/Integration.Vsix.UnitTests/ErrorList/SonarErrorListDataSource_IssueLocationStoreTests.cs

```diff
@@ -177,15 +177,53 @@ public void RefreshErrorList_FactoryIsNotRegistered_EventNotRaised()
         }
 
         [TestMethod]
-        public void LocationsUpdated_NullArg_Throws()
+        public void Refresh_NullArg_Throws()
         {
             var testSubject = CreateTestSubject();
             Action act = () => testSubject.Refresh(null);
             act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("affectedFilePaths");
         }
 
         [TestMethod]
-        public void LocationsUpdated_FactoriesWithMatchingFilesUpdatedAndErrorListRefreshed()
+        public void RefreshOnBufferChanged_NullArg_Throws()
+        {
+            var testSubject = CreateTestSubject();
+            Action act = () => testSubject.RefreshOnBufferChanged(null);
+            act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("affectedFilePath");
+        }
+
+        /// <summary>
+        /// Most of the behaviour of Refresh(file) and RefreshOnBufferChanged(files) is the same.
+        /// This delegate is used by tests that check the common functionality
+        /// i.e. there is a common internal test method that does the common setup and assertions.
+        /// This delegate is passed into that common method to make a call to either 
+        /// "Refresh(file)" or "RefreshOnBufferChanged(files)"
+        /// </summary>
+        private delegate void RefreshXXXTestOperation(SonarErrorListDataSource testSubject, string filePath);
+
+        private static void CallRefresh(SonarErrorListDataSource testSubject, string filePath)
+            => testSubject.Refresh(new[] { filePath });
+
+        private static void CallRefreshOnBufferChanged(SonarErrorListDataSource testSubject, string filePath)
+        {
+            // The RefreshOnBufferChanged method should never raise an IssuesChanged event, so we
+            // can add it here as an invariant check in all tests
+            var issueChangedHandler = CreateAndRegisterIssuesEventHandler(testSubject);
+
+            testSubject.RefreshOnBufferChanged(filePath);
+
+            CheckIssuesChangedEventIsNotRaised(issueChangedHandler);
+        }
+        [TestMethod]
+        public void Refresh_FactoriesWithMatchingFilesUpdatedAndErrorListRefreshed()
+            => RefreshXXX_FactoriesWithMatchingFilesUpdatedAndErrorListRefreshed(CallRefresh, shouldRaiseIssuesChanged: true);
+
+        [TestMethod]
+        public void RefreshOnBufferChanged_FactoriesWithMatchingFilesUpdatedAndErrorListRefreshed()
+            => RefreshXXX_FactoriesWithMatchingFilesUpdatedAndErrorListRefreshed(CallRefreshOnBufferChanged, shouldRaiseIssuesChanged: false);
+
+        private void RefreshXXX_FactoriesWithMatchingFilesUpdatedAndErrorListRefreshed(RefreshXXXTestOperation testOp,
+            bool shouldRaiseIssuesChanged)
```

Comment:
This is the only "common" test method that expects specific events to be raised.
I've added event checks.

None of the other tests have any functional changes to the body of the test.